### PR TITLE
Feature: add captcha for SignIn process

### PR DESCRIPTION
**Pull request by Kohelbekker**

### CODE REVIEW

**Review by akhlopiachyi — APPROVED** [review]

File: src/components/SignIn/SignIn.test.tsx [hunk 1]
```diff
@@ -21,6 +22,11 @@ const defaults: SignInProps = {
     handleChangeFocusField: jest.fn(),
     changePassword: jest.fn(),
     changeEmail: jest.fn(),
+    captchaType: 'none',
+    renderCaptcha: null,
+    reCaptchaSuccess: false,
+    geetestCaptchaSuccess: false,
```
Comment (akushniruk) [inline src/components/SignIn/SignIn.test.tsx:28]: can we combine reCaptchaSuccess and geetestCaptchaSuccess? (we can have only one type of captcha at the same time)

File: src/components/SignIn/index.tsx [hunk 2]
```diff
@@ -36,6 +38,11 @@ export interface SignInProps {
     handleChangeFocusField: (value: string) => void;
     changePassword: (value: string) => void;
     changeEmail: (value: string) => void;
+    captchaType?: 'recaptcha' | 'geetest' | 'none';
+    renderCaptcha?: JSX.Element | null;
+    reCaptchaSuccess?: boolean;
+    geetestCaptchaSuccess?: boolean;
```
Comment (akushniruk) [inline src/components/SignIn/index.tsx:44]: can we combine reCaptchaSuccess and geetestCaptchaSuccess? (we can have only one type of captcha at the same time)

File: src/components/SignIn/index.tsx [hunk 3]
```diff
@@ -94,6 +105,11 @@ const SignIn: React.FC<SignInProps> = ({
         [handleChangeFocusField]
     );
 
+    const isButtonDisabled = (): boolean => {
+        return (((captchaType === 'recaptcha' && !reCaptchaSuccess) ||
+        (captchaType === 'geetest' && !geetestCaptchaSuccess)) && captchaLogin()) ? true : false;
```
Comment (akushniruk) [inline src/components/SignIn/index.tsx:110]: if u will combine reCaptchaSuccess && geetestCaptchaSuccess please update this place too

File: src/screens/SignInScreen/index.tsx [hunk 4]
```diff
@@ -160,26 +200,29 @@ class SignIn extends React.Component<Props, SignInState> {
 
     private handleSignIn = () => {
         const { email, password } = this.state;
+        const { configs: { captcha_type }, captcha_response } = this.props;
 
-        this.props.signIn({
-            email,
-            password,
-        });
+        if ((captcha_type === 'recaptcha' || captcha_type === 'geetest') && captchaLogin()) {
```
Comment (akushniruk) [inline src/screens/SignInScreen/index.tsx:205]: are really we need to check both? maybe better if we do: 
``` if (captcha_type && captchaLogin())```

File: src/screens/SignInScreen/index.tsx [hunk 5]
```diff
@@ -160,26 +200,29 @@ class SignIn extends React.Component<Props, SignInState> {
...
             this.setState({
                 error2fa: 'Please enter 2fa code',
             });
         } else {
-            this.props.signIn({
-                email,
-                password,
-                otp_code: otpCode,
-            });
+            if ((captcha_type === 'recaptcha' || captcha_type === 'geetest') && captchaLogin()) {
```
Comment (akushniruk) [inline src/screens/SignInScreen/index.tsx:221]: same

File: src/screens/SignInScreen/index.tsx (hunk 4 above)
Comment (Kohelbekker) [reply src/screens/SignInScreen/index.tsx:205]: There is 3rd variant `captcha_type ===  'none'`, but I guess here can be `captcha_type !== 'none`

**Review by akushniruk — APPROVED** [review]

**Pull request merged**